### PR TITLE
fix(import-export): Use query document count in export COMPASS-4537, COMPASS-4906

### PR DESCRIPTION
**Pull request by Anemy**
COMPASS-4537 COMPASS-4906

This PR fixes a couple bugs around import/export:
- Fixes opening the export modal by properly counting documents on time-series collections and views. The collection function `estimatedDocumentCount` errors on views and time-series collections, so now we use `countDocuments`. This is more expensive, so we could change up the ui here to either background load with a loader, or hide the count entirely. We do get to skip the count when we open the export from the export button on `compass-crud` by passing the count from the resulting query there.
- Fixes showing the correct count for the query used to find the documents to be exported. Previously we would always show the estimated count, although we would export the expected amount of documents resulting from the query.
- Fixes a small bug around showing `undefined` for an empty query (screenshot in comment).
- Fixes opening the export collection modal from the app menu. Filed COMPASS-4911 which is a bug around using the app menu to open the import and export modals. This might be noticed more now that the export collection menu item works properly.
- Fixes double listeners to the `open-export` and `open-import` app registry messages/events. Previously we had both of these modals sharing the same store and subscribing to the same messages/events. Didn't really cause any issues before, but now that we're supplying the correct count we probably don't want to be duplicating a possibly expensive `countDocuments` call.

### CODE REVIEW

File: packages/compass-import-export/src/utils/get-shell-js.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import {stringify} from 'mongodb-query-parser';
+import { stringify } from 'mongodb-query-parser';
 import toNS from 'mongodb-ns';
 
 export default function(ns, spec) {
   let ret = `db.${toNS(ns).collection}.find(\n`;
-  ret += '  ' + stringify(spec.filter, '');
+  ret += '  ' + stringify(spec.filter ? spec.filter : {}, '');
```
Comment (Anemy) [inline packages/compass-import-export/src/utils/get-shell-js.js:6]: Fixes:
Screen Shot 2021-06-29 at 6 46 24 PM
Now:
Screen Shot 2021-06-29 at 6 48 40 PM

**Review by gribnoysup — APPROVED** [review]
Looks good

File: packages/compass-import-export/src/modules/export.js [hunk 2]
```diff
@@ -322,16 +323,51 @@ export const changeExportStep = (status) => ({
...
+    && !query.limit
+    && !query.skip
+  ) {
+    try {
+      const runEstimatedDocumentCount = promisify(dataService.estimatedCount.bind(dataService));
+      const count = await runEstimatedDocumentCount(ns, {});
+
+      return count;
+    } catch (estimatedCountErr) {
+      // `estimatedDocumentCount` is currently unsupported for
```
Comment (mcasimir) [inline packages/compass-import-export/src/modules/export.js:339]: A `debug` call here would be helpful

File: packages/compass-import-export/src/modules/export.js [hunk 3]
```diff
@@ -322,16 +323,51 @@ export const changeExportStep = (status) => ({
...
+
+  const runCount = promisify(dataService.count.bind(dataService));
+
+  const count = await runCount(
+    ns,
+    query.filter || {},
+    {
+      ...(query.limit ? { limit: query.limit } : {} ),
+      ...(query.skip ? { skip: query.skip } : {} )
+    }
```
Comment (mcasimir) [inline packages/compass-import-export/src/modules/export.js:353]: we need some kind of `maxTimeMs` and handle a failure somehow?

File: packages/compass-import-export/src/modules/export.js [hunk 4]
```diff
@@ -322,16 +323,51 @@ export const changeExportStep = (status) => ({
...
 
+const fetchDocumentCount = async(dataService, ns, query) => {
+  // When there is no filter/limit/skip try to use the estimated count.
+  if (
+    (!query.filter || Object.keys(query.filter).length < 1)
+    && !query.limit
+    && !query.skip
+  ) {
+    try {
+      const runEstimatedDocumentCount = promisify(dataService.estimatedCount.bind(dataService));
```
Comment (mcasimir) [inline packages/compass-import-export/src/modules/export.js:334]: Nice idea!

**Review by mcasimir — APPROVED** [review]
Wow the amount of fixes :)

**Pull request merged**